config/vufind/config.ini

```diff
@@ -1318,7 +1318,7 @@ url = "https://api.vlb.de/api/v1/cover/"
 [QRCode]
 ; This setting controls the image to display when no qrcode is available.
 ; The path is relative to the base of your theme directory.
-;noQRCodeAvailableImage = images/noQRCode.gif
+noQRCodeAvailableImage = images/noQRCode.gif
```

Comment:
Exactly, since "images/noQRCode.gif" was also the default in `QRCode/Loader.php`, this should not change anything.

Reply:
But the behavior of commenting out the setting has changed, hasn't it? If you do not provide the setting, then NOTHING will be displayed when a QR code is unavailable, correct? If that's the case, maybe we should just add another line of comment, like:

```
; If commented out, no image will be displayed for unavailable QR codes
```

(Again, sorry if I'm misunderstanding the logic here!)

Reply:
Yes, adding code to adjust this in VuFind\Config\Upgrader would definitely be a good idea!

Reply:
Unfortunately that does not work as I expected. If a setting is commented out in the local configs it will always be overwritten when upgrading. That means having a setting not commented out by VuFind's default and commenting it out for deactivation local does not really work.

We could either leave "noQRCodeAvailableImage" commented out in the config by default but by that change the default behavior. Or we could set `noQRCodeAvailableImage = false` for deactivating the setting instead of commenting it out.

What do you think?

Reply:
I suspect that this feature is rarely used, if ever, so it's probably not hugely impactful. I think a note in the changelog about the adjustment to default behavior is probably sufficient if there's no straightforward way to make the transition entirely seamless. I'll put a TODO comment on this to write that up when merging, and I'll do some more review and testing of this as soon as time permits.

Reply:
We have the same problem for `noCoverAvailableImage`. So we also would have to deactivate that by default or use `noCoverAvailableImage = false` for deactivation instead of commenting it out.

Reply:
Ahh, I see. I think I like the idea of supporting a false value, since that's a less backward-breaking change (if I'm thinking about this correctly).

Reply:
`false` already works for deactivating the setting. So, I just added a test for that and updated the comments in the config.

Reply:
Great, thank you!